resources/templates/md/posts/2016-04-04-state-management.md

```diff
@@ -0,0 +1,364 @@
+{:title "Clojure state management"
+:description  "How we can manage states in Clojure (how data might change over time), or llamas love candies too :)"
+:layout :post
+:tags  ["Clojure" "ClojureScript" "State management"]
+:toc false
+:author "Dajana Štiberová"
+:image ""}
+
+Long time ago, during the one interview, I was asked to describe the difference between `vars` and `atoms`.
+It was pretty simple question, but at that time it was really hard to describe this difference for me.
+Now I'm more skilled in Clojure but I'm still thinking that this is an interesting question and when
+I was looking for some topic for the blog post, I decided to use this one.
+
+So if you want to read something about state management, or you want to know how to dance like Greg the llama, this post is for you :)
+
+## States and Identities
+
+We need to know that in Clojure, there is the difference between the state and the identity.
+
+`States` are an immutable data.
+A state is simply the value of an identity at a particular time. State never changes.
+
+`Identities` unify different values over the time.
+Identity types are mutable references to immutable values,
+and can be modeled by using a reference types (`Vars`, `Atoms`, `Agents`, `Refs`)
+
+
+### *Example
+
+Our example is the best llama on earth, Greg the llama.
+Greg is an identity.
+Greg wants to go to the party with Gregorina. He is really good dancer, and he wants to teach us, how to dance.
+
+So, Greg is dancing and every snapshot of this action is one state.
+
+Greg has a series of different values (moves) over time (states 1-4).
+We need to realize, that we can transition the state, but it is not possible to change or modify the previous state.
+Identity (Greg) can be modeled by using some reference type pointing to the value,
+and subsequently calling apropriate functions to create a new states.
+
+NOTE:
+
+If you don't have any knowledge about threads in Clojure, you should read about threads first.
+
+## How we can manage states
+(or how the data might change over time)
```

Comment:
why this is on a newline? 
<img width="995" alt="screen shot 2016-04-11 at 19 47 34" src="https://cloud.githubusercontent.com/assets/15321488/14436742/5bd5a05a-001e-11e6-8897-6099a2d8ad78.png">